### PR TITLE
fix(ci): handle concurrent hotfix and release

### PR DESCRIPTION
**Pull request by tuntoja**
## Description

Handle concurrent hotfix and release branches by adding the two elements:
* delivery testing toward specific dirs (either /hotfix or /release for each components delivered)
* add PR_number to package name (cloud specific)

**Fixes** #MON-35276

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [ ] 23.10.x
- [x] 24.04.x (master)

### CODE REVIEW

File: .github/workflows/awie.yml [hunk 1]
```diff
@@ -27,170 +27,175 @@ jobs:
...
+  #         cache_key: ${{ github.sha }}-${{ github.run_id }}-deb-${{ matrix.distrib }}
+  #         stability: ${{ needs.get-version.outputs.stability }}
+  #         release_type: ${{ needs.get-version.outputs.release_type }}
+  #         release_cloud: ${{ needs.get-version.outputs.release_cloud }}
 
   promote:
     needs: [get-version]
-    if: ${{ contains(fromJson('["stable"]'), needs.get-version.outputs.stability)  && github.event_name != 'workflow_dispatch' }}
+    #if: ${{ contains(fromJson('["stable"]'), needs.get-version.outputs.stability)  && github.event_name != 'workflow_dispatch' }}
+    if: ${{ contains(fromJson('["stable"]'), needs.get-version.outputs.stability) && github.event_name == 'pull_request' && github.event.action == 'closed' && github.event.pull_request.merged == true }}
```
Comment (kduret) [inline .github/workflows/awie.yml:198]:
```suggestion
    if: ${{ contains(fromJson('["stable"]'), needs.get-version.outputs.stability) && github.event_name == 'pull_request' && github.event.pull_request.merged == true }}
```

File: .github/workflows/get-version.yml [hunk 2]
```diff
@@ -108,16 +118,34 @@ jobs:
...
+                echo "GITHUB_RELEASE_TYPE is: $GITHUB_RELEASE_TYPE"
+                echo "GITHUB_RELEASE_BRANCH_PR_NUMBER is: $GITHUB_RELEASE_BRANCH_PR_NUMBER"
+                echo "GITHUB_RELEASE_CLOUD is: $GITHUB_RELEASE_CLOUD"
+                # Github ouputs
+                echo "release=`echo $GITHUB_RELEASE_BRANCH_PR_NUMBER`.`date +%s`.`echo ${{ github.sha }} | cut -c -7`" >> $GITHUB_OUTPUT
+                echo "release_type=$GITHUB_RELEASE_TYPE" >> $GITHUB_OUTPUT
+                echo "release_cloud=$GITHUB_RELEASE_CLOUD" >> $GITHUB_OUTPUT
               else
                 echo "release=1" >> $GITHUB_OUTPUT
+                echo "release_cloud=0" >> $GITHUB_OUTPUT
```
Comment (kduret) [inline .github/workflows/get-version.yml:147]:
```suggestion
                echo "release_cloud=$GITHUB_RELEASE_CLOUD" >> $GITHUB_OUTPUT
```

File: .github/workflows/get-version.yml [hunk 3]
```diff
@@ -108,16 +118,34 @@ jobs:
...
+                echo "release_type=$RELEASE_TYPE" >> $GITHUB_OUTPUT
```
Comment (kduret) [inline .github/workflows/get-version.yml:148]:
```suggestion
                echo "release_type=$GITHUB_RELEASE_TYPE" >> $GITHUB_OUTPUT
```

File: .github/workflows/get-version.yml [hunk 4]
```diff
@@ -108,16 +118,34 @@ jobs:
               echo "release=1" >> $GITHUB_OUTPUT
               ;;
             release* | hotfix*)
-              # Handle workflow_dispatch run triggers
+              # Set default release values
```
Comment (kduret) [inline .github/workflows/get-version.yml:121]: you should set default values when it's not a release / hotfix too

File: .github/actions/promote-to-stable/action.yml [hunk 5]
```diff
@@ -34,22 +40,40 @@ runs:
...
+          ROOT_REPO_PATH="test-rpm-standard"
         fi
 
+        # Build search path based on release_cloud and release_type values
+        # if cloud, search in testing-<release_type> path
+        # if non-cloud, search in the testing usual path
+        if [[ ${{ inputs.release_cloud }} -eq 1 && ${{ inputs.release_type }} == "hotfix" ]] || [[ ${{ inputs.release_cloud }} -eq 1 && ${{ inputs.release_type }} == "release" ]];then
+          SEARCH_REPO_PATH="${{ inputs.major_version }}/${{ inputs.distrib }}/testing-${{ inputs.release_type }}/$ARCH/${{ inputs.module_name }}"
+        elif [[ ${{ inputs.release_cloud }} -eq 0 ]];then
+          SEARCH_REPO_PATH="${{ inputs.major_version }}/${{ inputs.distrib }}/testing/$ARCH/${{ inputs.module_name }}/"
```
Comment (kduret) [inline .github/actions/promote-to-stable/action.yml:68]: testing-release and testing-hotfix are not used for onprem ?

File: .github/docker/centreon-web/jammy/entrypoint/container.d/80-logs.sh
Comment (kduret) [inline .github/docker/centreon-web/jammy/entrypoint/container.d/80-logs.sh:1]: remove this file

File: .github/docker/centreon-web/jammy/scripts/autoinstall.php
Comment (kduret) [inline .github/docker/centreon-web/jammy/scripts/autoinstall.php:1]: remove this file

File: .github/actions/promote-to-stable/action.yml (hunk 5 above)
Comment (tuntoja) [reply .github/actions/promote-to-stable/action.yml:68]: will not be used for on prem
only cloud is involved in this

File: .github/workflows/awie.yml [hunk 6]
```diff
@@ -200,6 +204,20 @@ jobs:
...
+
+      #     # Check what was the merged branch (= last commit with "release(hotfix)" as a prefix)
+      #     echo "[DEBUG] MERGED_BRANCH is: $MERGED_BRANCH"
+      #     echo ""
+      #   shell: bash
+      #   id: get_merged_branch
+      #   env:
+      #     RELEASE_TYPE_CUSTOM: release_type_custom
+      #     RELEASE_CLOUD_CUSTOM: release_cloud_custom
+
```
Comment (sc979) [inline .github/workflows/awie.yml:220]: Kindly remove dead code.

File: .github/actions/rpm-delivery/action.yml [hunk 7]
```diff
@@ -64,21 +78,44 @@ runs:
...
+        elif [[ ${{ inputs.release_cloud }} -eq 0 ]];then
+          #ROOT_REPO_PATH="rpm-standard"
+          ROOT_REPO_PATH="test-rpm-standard"
+          UPLOAD_REPO_PATH="${{ inputs.version }}/${{ inputs.distrib }}/${{ inputs.stability }}/$ARCH/RPMS/${{ inputs.module_name }}/"
+        else
+          echo "Invalid combination of release_type and release_cloud"
+        fi
+
+        # Deliver based on inputs
+        #for ROOT_REPO_PATH in "rpm-standard" "rpm-standard-internal"; do
```
Comment (sc979) [inline .github/actions/rpm-delivery/action.yml:104]: kindly remove dead code

File: .github/actions/rpm-delivery/action.yml [hunk 8]
```diff
@@ -64,21 +78,44 @@ runs:
...
+        for ROOT_REPO_PATH in "test-rpm-standard" "test-rpm-standard-internal"; do
           for ARCH in "noarch" "x86_64"; do
             if [ "$(ls -A $ARCH)" ]; then
               if [ "${{ inputs.stability }}" == "stable" ]; then
-                jf rt upload "$ARCH/*.rpm" "$ROOT_REPO_PATH/${{ inputs.version }}/${{ inputs.distrib }}/${{ inputs.stability }}/$ARCH/RPMS/${{ inputs.module_name }}/" --flat
+                #jf rt upload "$ARCH/*.rpm" "$ROOT_REPO_PATH/${{ inputs.version }}/${{ inputs.distrib }}/${{ inputs.stability }}/$ARCH/RPMS/${{ inputs.module_name }}/" --flat
+                echo "[DEBUG] - Stability is ${{ inputs.stability }}, not delivering."
+              elif [ "${{ inputs.stability }}" == "testing" ]; then
+                #jf rt upload "$ARCH/*.rpm" "$ROOT_REPO_PATH/${{ inputs.version }}/${{ inputs.distrib }}/${{ inputs.stability }}/$ARCH/${{ inputs.module_name }}/${{ inputs.release_type }}/" --sync-deletes="$ROOT_REPO_PATH/${{ inputs.version }}/${{ inputs.distrib }}/${{ inputs.stability }}/$ARCH/${{ inputs.module_name }}/${{ inputs.release_type }}/" --flat
```
Comment (sc979) [inline .github/actions/rpm-delivery/action.yml:112]: kindly remove dead code

File: .github/actions/promote-to-stable/action.yml [hunk 9]
```diff
@@ -34,22 +40,40 @@ runs:
...
         # Cloud specific promote
         # delivery by default to onprem, override to internal if base branch is master
         if [[ ${{ inputs.github_base_ref }} == "master" ]]; then
-          ROOT_REPO_PATH="rpm-standard-internal"
+          #ROOT_REPO_PATH="rpm-standard-internal"
+          ROOT_REPO_PATH="test-rpm-standard-internal"
         else
-          ROOT_REPO_PATH="rpm-standard"
+          #ROOT_REPO_PATH="rpm-standard"
+          ROOT_REPO_PATH="test-rpm-standard"
```
Comment (sc979) [inline .github/actions/promote-to-stable/action.yml:59]: dead code ?

**Review by kduret — APPROVED** [review]

**Review by sc979 — APPROVED** [review]

**Pull request merged**